NineChronicles.Standalone.Executable/AWSSink.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon;
+using Amazon.CloudWatchLogs;
+using Amazon.CloudWatchLogs.Model;
+using Amazon.Runtime;
+using JetBrains.Annotations;
+using Nito.AsyncEx;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace NineChronicles.Standalone.Executable
+{
+    public class AWSSink : ILogEventSink, IDisposable
+    {
+        private readonly AmazonCloudWatchLogsClient _client;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+
+        private AsyncProducerConsumerQueue<LogEvent> _queue;
+
+        public AWSSink(AWSCredentials credentials, RegionEndpoint regionEndPoint, Func<string> logGroupNameGetter, Func<string> logStreamNameGetter = null)
+        {
+            Config = new AmazonCloudWatchLogsConfig();
+            Config.RegionEndpoint = regionEndPoint;
+            _queue = new AsyncProducerConsumerQueue<LogEvent>();
+            _client = new AmazonCloudWatchLogsClient(credentials, Config);
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            LogGroupNameGetter = logGroupNameGetter;
+            LogStreamNameGetter = logStreamNameGetter;
+
+            Worker(_cancellationTokenSource.Token);
+        }
+
+        public void Dispose()
+        {
+            _client.Dispose();
+            _cancellationTokenSource.Cancel();
```

Comment:
Is it safe we perform this cancellation after `_client.Dispose()`?

Reply:
Applied at e61665925004f08d3df6940fc92021d49023926e